### DISCUSSION TITLE
Device compatibility M1065

### DISCUSSION THREAD

**Discussion by ghost** [AxisCommunications/discussions#252] [ACAP]
Can I write ACAP applications for a M1068, firmware version 9.80.28?

**Comment by daniel-falk** [comment]
Hi @qolson5,

Yes, you can. SDK 3 is still supported and is compatible with FW back to 9.70 [as seen in this table](https://help.axis.com/en-us/acap-3-developer-guide#find-the-right-sdk-for-software-compatibility). Make sure to use the minor 3.1 (or 3.0), otherwise you will get issues with libc and glib compatibility. You also need to set the [Required embedded development version](https://help.axis.com/en-us/acap-3-developer-guide#manifest-file-content) to 2.16 if I remember right to make the application installable in the 9.80 FW.

A few things to note:
* SDK 3 will probably become deprecated at some point (as far as I know there is no time frame available from Axis) - it is probably worth to spend some time making sure the application is as compatible as possible with Native SDK 4 too, otherwise migration will be expensive.
* SDK 3.0 to 3.2 does not have the json-based manifest file and the `acap-build` script, this was added in SDK 3.3
* Some libraries like axhttp and axparams are removed in SDK 4, try to avoid using these. You can e.g. use the monkey webserver which works with both SDK 3 and SDK 4.

I hope this get's you started. Tell me if you want more information on how to keep the ACAPs portable and compatible with both SDKs, then I can send you some links.

**Comment by kawamata-teruyasu** [comment]
@daniel-falk,

> Tell me if you want more information on how to keep the ACAPs portable and compatible with both SDKs, then I can send you some links.

Could you please share the information on this? I'm thinking of my app to be build with SDK v3 and v4. Especially, I'm wondering how to have common implementation that serves web pages for application settings. The monkey webserver you mentioned in old post seems a solution for this, but any other information are appreciated. Thanks.
